js/initcontext.go

```diff
@@ -140,14 +150,52 @@ func (i *InitContext) Require(arg string) goja.Value {
 	}
 }
 
+type moduleInstanceCoreImpl struct {
+	ctxPtr *context.Context
+	// we can technically put lib.State here as well as anything else
+}
+
+func (m *moduleInstanceCoreImpl) GetContext() context.Context {
+	return *m.ctxPtr
+}
+
+func (m *moduleInstanceCoreImpl) GetInitEnv() *common.InitEnvironment {
+	return common.GetInitEnv(*m.ctxPtr) // TODO thread it correctly instead
+}
+
+func (m *moduleInstanceCoreImpl) GetState() *lib.State {
+	return lib.GetState(*m.ctxPtr) // TODO thread it correctly instead
+}
+
+func (m *moduleInstanceCoreImpl) GetRuntime() *goja.Runtime {
+	return common.GetRuntime(*m.ctxPtr) // TODO thread it correctly instead
+}
+
+func toESModuleExports(exp modules.Exports) map[string]interface{} {
+	result := make(map[string]interface{}, len(exp.Named)+2)
+
+	for k, v := range exp.Named {
+		result[k] = v
+	}
+	// Maybe check that those weren't set
+	result["default"] = exp.Default
+	result["__esModule"] = true // this so babel works with
```

Comment:
Could you expand on this comment? Why is this required exactly?

It shouldn't be necessary for ES5 (`--compatibility-mode=base`), so it feels hacky.